src/VocabGamePlay.jsx

```diff
@@ -1,48 +1,62 @@
 import React, { Component } from 'react';
 import "./VocabGamePlay.scss";
-import {GameDataset} from "./VocabGameData.js";
+import { GameDataset } from "./VocabGameData.js";
+import WordCard from './WordCard';
 
 class VocabGamePlay extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            firstSelection: null,
-            secondSelection: null,
-            vocabDataSet: GameDataset,
-        }
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstSelection: null,
+      secondSelection: null,
+      vocabDataSet: GameDataset,
     }
-    handleClick(e, data) {
-      console.log(data);
-      data.isSelected = !data.isSelected;
-      if (this.state.firstSelection === null) {
+  }
+
+  handleClick(e, data) {
+    console.log(data);
+    data.isSelected = !data.isSelected;
+    if (this.state.firstSelection === null) {
+      this.setState({ firstSelection: data });
+    } else if (this.state.secondSelection === null) {
+      if (data === this.state.firstSelection) {
         this.setState({ firstSelection: data });
+        return;
       }
-      else if (this.state.secondSelection === null) {
-        if (data === this.state.firstSelection) {
-          this.setState({ firstSelection: data });
-          return;
-        }
-        this.setState({ secondSelection: data }, () => { this.checkMatch(this.state.firstSelection, this.state.secondSelection) });
-      }
-    }
-    checkMatch(first, second){
-       this.isCorrectUpdate(first.word===second.translation);
+      this.setState({ secondSelection: data }, () => { this.checkMatch(this.state.firstSelection, this.state.secondSelection) });
     }
+  }
+  
 
-    isCorrectUpdate(isMatch){
-        let first=this.state.firstSelection;
-        let second=this.state.secondSelection;
-        first.isCorrect=isMatch;
-        second.isCorrect=isMatch;
+  checkMatch(first, second) {
+    this.isCorrectUpdate(first.word === second.translation);
+  }
+
+  isCorrectUpdate(isMatch) {
+    let first = this.state.firstSelection;
+    let second = this.state.secondSelection;
+    first.isCorrect = isMatch;
+    second.isCorrect = isMatch;
 
-        setTimeout(()=> {this.setState({
-            firstSelection: null,
-            secondSelection: null,
-        })}, 2000);
+    setTimeout(() => {
+      this.setState({
+        firstSelection: null,
+        secondSelection: null,
+      })
+    }, 2000);
+  }
+
+  renderGameCards = (mappedObject) => {
+    let renderedItems = []
+    for (var value of mappedObject.values()) {
+      renderedItems.push(<WordCard key={value.word} data={value} />)
     }
-    
+    return renderedItems;
+  }
 
   render() {
+    let { vocabDataSet } = this.state;
```

Comment:
Can someone explain what this line is doing? Thanks!

Reply:
its deconstructing the `this.state` object.

its the same as using `this.state.vocabDataSet`. But I put this here in case we will be using vocabDataSet multiple times because we dont want to have to use `this.state.vocabDataSet` everytime. It's also so we can easily add other variables from `this.state`

aka:
`let { vocabDataSet, firstSelection, secondSelection } = this.state`

Reply:
Errrrrrrrrr yeah I don't follow, sorry! But let me ask this then: the next line's `console.log`, is that logging the `let {vocabDataSet}`?